### ISSUE TITLE
Introduce TrilinosWrappers::make_rcp(...).

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#16202]
Looking at #16156, we are introducing a number of places again where we're using bare `operator new` calls. We've tried to replace as many of these by `std::make_shared()` and `std::make_unique()` as possible in recent years to make it easier to reason about whether a piece of code may or may not have memory leaks. We should create a function
```
template<class T, class... Args>
Teuchos::RCP<T>
make_rcp(Args&&... args)
{
    return Teuchos::RCP<T>(new T(std::forward<Args>(args)...));
}
```
and use it wherever we allocate memory via direct `new` calls right now.

(Of course, it would be nice if Teuchos already had such a function for us, but I can't find one.)

@kinnewig FYI.

**Comment by maartenarnst** [comment]
There is such a function :)
- https://github.com/trilinos/Trilinos/pull/10647

**Comment by bangerth** [comment]
@maartenarnst Nice! Do you happen to know which release this was introduced in?

**Comment by maartenarnst** [comment]
The PR with this function was merged in June 2022. It was during a period in which the Trilinos team seems to have updated the version number less frequently. The earliest version in which this PR is included seems to be 14.0 from May 2023.

**Comment by bangerth** [comment]
Fixed by #16270.
